auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -438,6 +450,26 @@ public T setAvailableProviders(@NonNull List<IdpConfig> idpConfigs) {
                 } else {
                     mProviders.add(config);
                 }
+
+                if (config.getProviderId().equals(FACEBOOK_PROVIDER)) {
+                    try {
+                        Class c = com.facebook.FacebookCallback.class;
+                    } catch (NoClassDefFoundError e) {
+                        throw new RuntimeException("Facebook provider cannot be configured " +
+                               "without dependency. Did you forget to add " +
+                               "'com.facebook.android:facebook-android-sdk:VERSION' dependency?");
+                    }
+                }
+
+                if (config.getProviderId().equals(TWITTER_PROVIDER)) {
+                    try {
+                        Class c = com.twitter.sdk.android.core.TwitterCore.class;
```

Comment:
We're actually only depending on Twitter core now (which means we don't pull in Fabric).  This is new with the upgrade to Twitter SDK 3.0